### PR TITLE
dw-dma: make sure current pointer values are always consistent

### PR DESCRIPTION
**Pull request by lyakh**
Incrementing the current DMA position pointer isn't atomic, use a
channel spin-lock to synchronise its reads and writes.

### CODE REVIEW

File: src/drivers/dw/dma.c [hunk 1]
```diff
@@ -1073,7 +1076,7 @@ static int dw_dma_get_data_size(struct dma_chan_data *channel,
 	tr_dbg(&dwdma_tr, "dw_dma_get_data_size(): dma %d channel %d get data size",
 	       channel->dma->plat_data.id, channel->index);
 
-	irq_local_disable(flags);
+	spin_lock_irq(&channel->dma->lock, flags);
```
Comment (plbossart) [inline src/drivers/dw/dma.c:1079]: are we allowed to take a spinlock while writing a trace in dw_dma_avail_data_size() and dw_dma_free_data_size()

Comment (paulstelian97) [reply src/drivers/dw/dma.c:1079]: DMA trace will go asynchronously and etrace works without any internal spinlocks AFAIK. I don't think this is an issue.

Comment (lyakh) [reply src/drivers/dw/dma.c:1079]: @plbossart this is a per-channel spin-lock, and as far as I understand DW DMA isn't used for writing traces? So there should be no recursion?

Comment (lgirdwood) [reply src/drivers/dw/dma.c:1079]: DW DMA is used for trace, but trace DMA work is scheduled so should not deadlock here.

Comment (plbossart) [reply src/drivers/dw/dma.c:1079]: ok then.

**Comment by lgirdwood** [comment]
Jenkins DUT failure.

**Pull request merged**